_overviews/collections-2.13/conversion-between-option-and-collection.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: multipage-overview
+title: Conversion Between Option and Collection
+partof: collections-213
+overview-name: Collections
+
+num: 18
+previous-page: conversion-between-java-and-scala-collections
+
+permalink: /overviews/collections-2.13/:title.html
+---
+`Option` can be seen conceptually as a collection that has zero or exactly one element, and can be iterated once or more than one time. `Option` does provide a degree of interoperability with the collection types found in the package `scala.collection`, in particular, it implements the interface `IterableOnce`, which models the simplest form of collections. However `Option` is not a full collections in the sense of implementing the full `Iterable` interface, because although an iterator of zero to one elements exists for `Option`, we cannot provide a sensible implementation for [`fromSpecific`](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/collection/Iterable.scala#L173) which turns one collection into another
+
+
+```
+def fromSpecific(coll: IterableOnce[A]): C
+```
+
+For example if we attempted to defined it as `coll.headOption`, then we would throw away some elements when `coll` has more than one element. For this reason, starting [Scala 2.13](https://github.com/scala/scala/pull/8038), `Option` was made an `IterableOnce` but not an `Iterable`. To help clarify the difference between the two, mentally rename `Iterable` to `Collection` whilst `IterableOnce` becomes `Iterable`.
+
+Hence `Option` can be used everywhere an `IterableOnce` is expected, for example, when calling `flatMap` on a collection (or inside a for-comprehension)
```

Comment:
I would propose to go less into the technical details. I think what is important is to let the reader what is supported, but I don’t think we should explain much what it would entail to make `Option` implement `Iterable`. (Maybe others disagree?)
Also, I don’t think it helps to propose a mental model where `Iterable` is renamed to `Collection` and `IterableOnce` is renamed to `Iterable`.

```suggestion
`Option` can be seen as a collection that has zero or exactly one element, and it provides a degree of interoperability with the collection types found in the package `scala.collection`. In particular, it implements the interface `IterableOnce`, which models the simplest form of collections: something that can be iterated over, at least once. However, `Option` does not implement the more comprehensive interface of `Iterable`. Indeed, we cannot provide a sensible implementation for the operation [`fromSpecific`](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/collection/Iterable.scala#L173), which is supposed to create an `Option` from a collection of possibly more than one element. Starting from [Scala 2.13](https://github.com/scala/scala/pull/8038), `Option` was made an `IterableOnce` but not an `Iterable`.

Hence `Option` can be used everywhere an `IterableOnce` is expected, for example, when calling `flatMap` on a collection (or inside a for-comprehension)
```